src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php

```diff
@@ -152,6 +152,16 @@ public function testClientError()
         $this->assertSame(404, $response->getStatusCode());
         $this->assertSame(['application/json'], $response->getHeaders(false)['content-type']);
         $this->assertNotEmpty($response->getContent(false));
+
+        $response = $client->request('GET', 'http://localhost:8057/404');
```

Comment:
could we create separate tests rather than a single test `testClientError` actually performing multiple subsequent tests ? This would allow each test name to describe what it is testing.